### PR TITLE
Add configured OEmbed query parameters to all OEmbed endpoints

### PR DESCRIPTION
**Pull request by digilist**
I noticed that oEmbed query parameters are not passed to all oEmbed endpoints (in my case I noticed this specifically with Vimeo).

I tried to come up with a fix. I am not sure if that's the right solution to the problem or introduces different problems. But at least it fixes it for Vimeo and doesn't break any tests ;)

I am looking forward to your feedback.

### CODE REVIEW

File: src/OEmbed.php [hunk 1]
```diff
@@ -53,11 +53,22 @@ protected function detectEndpoint(): ?UriInterface
...
-            ?: $this->detectEndpointFromProviders();
+            ?: null;
+
+        if ($endpoint === null) {
+            return $this->detectEndpointFromProviders();
+        }
+
+        // Add configured OEmbed query parameters
+        parse_str($endpoint->getQuery(), $query);
+        $query = http_build_query(array_merge($query, $this->getOembedQueryParameters($query['url'])));
```
Comment (digilist) [inline src/OEmbed.php:68]: I used the `array_merge` here to keep the original order of the parameters.

I don't think it's really necessary, but changing the parameters order invalidates some cached files and breaks some tests. If this solution is okay, I can also rename old cache files (I have saved myself that work for now).

Comment (oscarotero) [reply src/OEmbed.php:68]: This overrides the query parameters of `$query` with the values returned by `getOembedQueryParameters` (including the values `url` and `format`). In think it's not needed to use this function, we can merge it directly with the settings:

```php
$query = array_merge($query, $this->extractor->getSetting('oembed:query_parameters') ?? []);
```

Comment (digilist) [reply src/OEmbed.php]: Makes sense, I adjusted it. Thanks for the quick review!

**Comment by oscarotero** [comment]
Thanks for your contribution!

**Pull request merged**